alembic/script/revision.py

```diff
@@ -964,6 +1032,13 @@ def verify_rev_id(cls, revision):
     def __init__(
         self, revision, down_revision, dependencies=None, branch_labels=None
     ):
+        if down_revision and revision in down_revision:
+            raise LoopDetected("Self-loop is detected on %s" % revision)
```

Comment:
**mike bayer** (zzzeek) wrote:

I think as LoopDetected/DependencyLoopDetected are constructed, these should just be given the "revision" and not a whole message right?

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/2379